src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -116,15 +116,24 @@ struct patcher
     unknown_symbol_kind = 8
   };
 
-  symbol_type m_symbol_type;
+   enum class buf_type {
+       ctrltext = 0,   // control code
+       ctrldata,       // control packet
+       preempt_save,   // preempt_save
+       preempt_restore // preempt_restore
+  };
+
+  buf_type m_buf_type = buf_type::ctrltext;
+  symbol_type m_symbol_type = symbol_type::shim_dma_48;
```

Comment:
```suggestion
  auto m_buf_type = buf_type::ctrltext;
  auto m_symbol_type = symbol_type::shim_dma_48;
```

Reply:
These two lines declare two member variables in struct patcher and initialize them.
Keyword "auto " cannot be used there.

Reply:
Ah, of course, I missed the context.